test/unit/outFile.spec.ts

```diff
@@ -0,0 +1,152 @@
+    [
+        "Import chain with a different order",
+        {
+            "main.ts": `
+                import { value } from "./reexport";
+                if (value !== true) {
+                    throw "Failed to import value";
+                }
+            `,
+            "export.ts": exportValueSource,
+            "reexport.ts": reexportValueSource,
+        },
+    ],
+    [
+        "Import diamond export -> reexport1 & reexport2 -> main",
+        {
+            "main.ts": `
+                import { value as a } from "./reexport1";
+                import { value as b } from "./reexport2";
+                if (a !== true || b !== true) {
+                    throw "Failed to import a or b";
+                }
+            `,
+            "export.ts": exportValueSource,
+            "reexport1.ts": reexportValueSource,
+            "reexport2.ts": reexportValueSource,
+        },
+    ],
+    [
+        "Import diamond different order",
+        {
+            "reexport1.ts": reexportValueSource,
+            "reexport2.ts": reexportValueSource,
+            "export.ts": exportValueSource,
+            "main.ts": `
+                import { value as a } from "./reexport1";
+                import { value as b } from "./reexport2";
+                if (a !== true || b !== true) {
+                    throw "Failed to import a or b";
+                }
+            `,
+        },
+    ],
+    [
+        "Modules in directories",
+        {
+            "main.ts": `
+                import { value } from "./module/module";
+                if (value !== true) {
+                    throw "Failed to import value";
+                }
+            `,
+            "module/module.ts": `
+                export const value = true;
+            `,
+        },
+    ],
+    [
+        "Modules aren't ordered by name",
+        {
+            "main.ts": `
+                import { value } from "./a";
+                if (value !== true) {
+                    throw "Failed to import value";
+                }
+            `,
+            "a.ts": `
+                export const value = true;
+            `,
+        },
+    ],
+    [
+        "Modules in directories",
+        {
+            "main/main.ts": `
+                import { value } from "../module";
+                if (value !== true) {
+                    throw "Failed to import value";
+                }
+            `,
+            "module.ts": `
+                export const value = true;
+            `,
+        },
+    ],
+    [
+        "LuaLibs are usable",
+        {
+            "module.ts": `
+                export const array = [1, 2];
+                array.push(3);
+            `,
+            "main.ts": `
+                import { array } from "./module";
+                if (array[2] !== 3) {
+                    throw "Array's third item is not three";
+                }            
+            `,
+        },
+    ],
+])("outFile tests (%s)", (_, files) => {
```

Comment:
I think it's better to have individual tests instead, common code can be extracted to a function